sdk/storage/azure-storage-blob/blob_service_client.cpp

```diff
@@ -0,0 +1,65 @@
+namespace azure { namespace storage { namespace blobs {
+
+class blob_service_client
+{
+    // TODO: Storage team is actually not happy with client level pipeline. We prefer request level pipeline.
+public:
+//ctor
+    //shared_key auth
+    blob_service_client(
+        azure::core::http::uri uri,
+        storage_shared_key_credential credential,
+        blob_client_options options
+        );
+
+    //token auth
+    blob_service_client(
+        azure::core::http::uri uri,
+        token_credential credential,
+        blob_client_options options
+        );
+
+    //connection string
+    blob_service_client(const std::string& connection_string, blob_client_options options);
+
+    //anonymous/SAS/Customized pipeline auth
+    blob_service_client(azure::core::http::uri service_uri, blob_client_options options);
+
+//static
+    static blob_container_client get_blob_container_client(std::string container_name);
+
+//APIs
+    azure::core::iterable<blob_container> list_blob_containers(
+        const std::string& prefix = "",
+        const std::string& continuation_token = "",
+        const uint16_t maximum_result = 100,
+        const bool include_metadata = true,
+        const azure::core::cancellation_token& cancellation_token = azure::core::cancellation_token::none()
+        ) const;
+
+    azure::core::http::response<user_delegation_key> get_user_delegation_key(
+        const std::chrono::time_point& end_time,
+        const std::chrono::time_point& start_time = std::chrono::steady_clock::now()
+        const azure::core::cancellation_token& cancellation_token = azure::core::cancellation_token::none()
+        ) const;
+
+    azure::core::http::response<blob_container_client> create_blob_container(
+        const std::string& blob_container_name,
+        const blob_container_access_type& public_access_type = blob_container_access_type::none(),
+        const storage_metadata& metadata = storage_metadata::none(),
+        const azure::core::cancellation_token& cancellation_token = azure::core::cancellation_token::none()
+        ) const;
+
+        azure::core::http::response<void> delete_blob_container(
+        const std::string& blob_container_name,
+        const blob_lease& lease = blob_lease::none(),
+        const azure::core::cancellation_token& cancellation_token = azure::core::cancellation_token::none()
+        ) const;
+
+private:
+    azure::core::http::uri m_service_uri;
+    azure::core::pipeline m_pipeline;
+    std::string m_account_name;
+};
+
+}}}
```

Comment:
Should we make sure these end in newlines?

Reply:
Yes :)